sample_metadata/parser/generic_parser.py

```diff
@@ -172,40 +231,147 @@ def get_sample_id(self, row: Dict[str, Any]) -> str:
         """Get external sample ID from row"""
 
     @abstractmethod
-    def get_sample_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, Any]:
+    async def get_sample_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, Any]:
         """Get sample-metadata from row"""
 
     @abstractmethod
-    def get_sequence_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, Any]:
+    async def get_sequence_meta(
+            self, sample_id: str, row: GroupedRow
+    ) -> Dict[str, Any]:
         """Get sequence-metadata from row"""
 
-    def get_analyses(
-        self, sample_id: str, row: GroupedRow, cpg_id: Optional[str]
+    async def get_analyses(
+            self, sample_id: str, row: GroupedRow, cpg_id: Optional[str]
     ) -> List[AnalysisModel]:
         """
         Get analysis objects from row. Optionally, a CPG ID can be passed for
         to help finding files for previously added samples that were already renamed.
         """
         return []
 
-    def get_qc_meta(self, sample_id: str, row: GroupedRow) -> Optional[Dict[str, Any]]:
+    async def get_qc_meta(
+            self, sample_id: str, row: GroupedRow
+    ) -> Optional[Dict[str, Any]]:
         """Get qc-meta from row, creates a Analysis object of type QC"""
         return None
 
-    def get_sample_type(self, sample_id: str, row: GroupedRow) -> Union[str, SampleType]:
+    def get_sample_type(
+            self, sample_id: str, row: GroupedRow
+    ) -> Union[str, SampleType]:
         """Get sample type from row"""
         return self.default_sample_type
 
-    def get_sequence_type(self, sample_id: str, row: GroupedRow) -> Union[str, SequenceType]:
+    def get_sequence_type(
+            self, sample_id: str, row: GroupedRow
+    ) -> Union[str, SequenceType]:
         """Get sequence type from row"""
         return self.default_sequence_type
 
-    def get_sequence_status(self, sample_id: str, row: GroupedRow) -> Union[str, SequenceStatus]:
+    def get_sequence_status(
+            self, sample_id: str, row: GroupedRow
+    ) -> Union[str, SequenceStatus]:
         """Get sequence status from row"""
         return self.default_sequence_status
 
-    def parse_manifest(  # pylint: disable=too-many-branches
-        self, file_pointer, delimiter=',', confirm=False, dry_run=False
+    async def process_group(
+            self,
+            rows: GroupedRow,
+            external_sample_id: str,
+            cpg_sample_id: str,
+            sequence_id: str,
```

Comment:
A bit pedantic :) (nice to be explicit that we might no have internal IDs at this stage)
```suggestion
            cpg_sample_id: Optional[str],
            sequence_id: Optional[str],
```